csharp/private/rules/imports.bzl

```diff
@@ -119,14 +124,20 @@ def _import_multiframework_library_impl(ctx):
         if attr != None:
             providers[tfm] = attr[CSharpAssembly[tfm]]
 
-    fill_in_missing_frameworks(providers)
+    if ctx.attr.version_smearing:
+        fill_in_missing_frameworks(providers)
```

Comment:
I kinda feel like `import_multiframework_library` shouldn't ever be doing this? I'm not sure. I'm thinking it's strategy should moreso be "take `CSharpAssembly[][]` and flatten it to `CSharpAssembly[]`, holes are as filled as the providers fill them.

Reply:
Iiii dunno though. It's just our strategy of doing the smearing to get dependency-across-version support coming back to bite us I guess :P